### PR TITLE
escape escape string

### PR DESCRIPTION
**Pull request by cryogenian**
Fixes https://github.com/precog/quasar-plugin-oracle/issues/9 
(well with https://github.com/precog/quasar/pull/4759)

### CODE REVIEW

**Comment by cryogenian** [comment]
@rintcius 

Also made `JdbcDiscovery` a trait instead of final class.

**Comment by cryogenian** [comment]
Also don't raise an error if there is no tables in schema

File: core/src/main/scala/quasar/lib/jdbc/JdbcDiscovery.scala [hunk 1]
```diff
@@ -229,6 +90,162 @@ object JdbcDiscovery {
...
+          case Ior.Right(table) =>
+            ("", escapeForSearch(escape, table.asString))
+
+          // The table in the specified schema
+          case Ior.Both(schema, table) =>
+            (escapeForSearch(escape, schema.asString), escapeForSearch(escape, table.asString))
+        }
+      }
+
+    private def escapeForSearch(escapeString: String, patternLiteral: String): String = {
```
Comment (rintcius) [inline core/src/main/scala/quasar/lib/jdbc/JdbcDiscovery.scala:241]: We should have tests to enforce that this kind of logic works as expected and keeps working as expected in all concrete dbms-es.

Not sure if we need to do this now (and for v2), but good to keep in mind for v3 at least.

Comment (cryogenian) [reply core/src/main/scala/quasar/lib/jdbc/JdbcDiscovery.scala:241]: I'd vote to put it as is for now, and refactor it when we need for v3. I.e. extract escape code somewhere and make `selectedTables` not working without it, like 
```scala
trait Escaper[+F[_]] {
  def apply(i: String): F[String]
}
object Escaper {
  def jdbc: Escaper[ConnectionIO] = ???
  def apply(escape: String): Escaper[Nothing] = ???
}
// in JdbcDiscovery
def apply(discoverableTypes: ..., escaper: Escaper[ConnectionIO]): ConnectionIO
```

**Review by rintcius — APPROVED** [review]

**Pull request merged**